cpp/dolfinx/mesh/Topology.cpp

```diff
@@ -1411,3 +1411,85 @@ mesh::entities_to_index(const Topology& topology, int dim,
   return indices;
 }
 //-----------------------------------------------------------------------------
+std::vector<std::vector<std::int32_t>>
+mesh::compute_mixed_cell_pairs(const Topology& topology,
+                               mesh::CellType facet_type)
+{
+  int tdim = topology.dim();
+  std::vector<mesh::CellType> cell_types = topology.entity_types(tdim);
+  std::vector<mesh::CellType> facet_types = topology.entity_types(tdim - 1);
```

Comment:
Do we want to generalise this for other sub-entities? I think all we'd need to change is replacing `tdim - 1` with `dim(facet_type)` everywhere and update the docstring